fft/src/fftwrap.rs

```diff
@@ -48,79 +33,46 @@ use crate::smallft::*;
    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 */
-#[no_mangle]
-pub unsafe extern "C" fn spx_fft_init(mut size: c_int) -> *mut c_void {
-    let mut table = Box::new(drft_lookup::new(size as usize));
-
-    return Box::into_raw(table) as *mut c_void;
-}
-
-#[no_mangle]
-pub unsafe extern "C" fn spx_fft_destroy(mut table: *mut c_void) {
-    let table = table as *mut drft_lookup;
-
-    Box::from_raw(table);
+pub fn spx_fft_init(size: usize) -> DrftLookup {
```

Comment:
This has a different return type: now the type lives on the stack, before it was allocated on the heap. Being a `pub fn`, this should maintain the same basic behaviour, even because it is pretty redundant with `DrftLookup::new`.

Reply:
We could integrate all `fftwrap` and `smallft` public methods into the `DrftLookup` struct. What do you think? In this way we can use only the `DrftLookup` struct to call all our public methods.

Reply:
If you do that now, you need to do a refactory where the functions are used. Maybe it could be easier to make use of this crate inside speexdsp-rs before changing the API, what do you think?

Reply:
No problem for the refactory. I'd rather change the API now since this crate isn't released yet and it's still not used inside `speexdsp-rs`. I could add all those changes in a new commit of this PR.